ProfileView/src/index.ts

```diff
@@ -0,0 +1,253 @@
+        this.requestUpdate();
+      })
+      .catch((err: Error) => {
+        console.log(err);
+        this.showSpinner = false;
+        this.requestUpdate();
+      });
+  }
+
+  // result from api is split and stored back to the input template.
+  // This is because the api does not return reliable template back.
+  setTaggedResults() {
+    const PRESET_TAGS = ["name", "email"];
+
+    PRESET_TAGS.forEach((x: string) => {
+      let matches = this.profile.filter((y: any) => y.query.tag === x);
+      if (x === "name") {
+        matches = matches.sort((a: any) => {
+          if (a.query.Metadata === "firstName") {
+            return -1;
+          } else if (a.query.Metadata === "lastName") {
+            return 1;
+          } else return 0;
+        });
+        // latest first name & last name
+        this.presetTags["name"] = [matches[0].result[0], matches[1].result[0]];
+      } else if (x === "email") {
+        this.presetTags["email"] = matches.map((y: any) => y.result).join(", ");
+      } else {
+        this.presetTags[x] = matches.map((y: any) => y.result);
+      }
+    });
+  }
+
+  getFormattedProfile() {
+    return html`
+      <div class="profile-bound default-template">
+        <cjaas-profile
+          .profile=${this.profile}
+          .presetTags=${this.presetTags}
+        ></cjaas-profile>
+        <section class="customer-journey" title="Customer Journey">
+          <div class="header inner-header">
+            <h4>Customer Journey</h4>
+          </div>
+          ${this.getTabs()}
+        </section>
+      </div>
+    `;
+  }
+
+  getLoadingSpinner() {
+    return this.showSpinner
+      ? html`
+          <div class="spinner-container">
+            <md-spinner size="20"></md-spinner>
+          </div>
+        `
+      : html`
+          <md-button
+            .circle=${true}
+            color="white"
+            size="28"
+            title="Reload Profile"
+            @click=${() => this.getProfile()}
+          >
+            <img
+              height="20px"
+              src="https://cjaas.cisco.com/web-components/icons/refresh_24.svg"
+            />
+          </md-button>
+        `;
+  }
+
+  getTabs() {
+    // tab data should return the event as such.. Should be rendered by stream component.
+    const tabs = this.profile.filter((x: any) => x.query.type === "tab");
+    // TODO: Track the selected tab to apply a class to the badge for color synching, making blue when selected
+    const activityTab = this.authToken
+      ? html`
+          <md-tab slot="tab">
+            <span>All</span>
+          </md-tab>
+          <md-tab-panel slot="panel">
+            <cjaas-timeline
+              .streamId=${this.authToken}
+              .filter=${`person eq '${this.customer}'`}
+              type="journey-and-stream"
+            ></cjaas-timeline>
+          </md-tab-panel>
+        `
+      : html`
+          <div class="center full-height">
+            <div>No data to show</div>
```

Comment:
This sort of stuff needs to be localized (or localizable...) can you add a Jira to keep that in mind?

Reply:
Ah, just to be clear, you mean the message text itself?

Reply:
yes